app/src/main/java/org/simple/clinic/teleconsultlog/medicinefrequency/MedicineFrequency.kt

```diff
@@ -1,8 +1,69 @@
 package org.simple.clinic.teleconsultlog.medicinefrequency
 
-enum class MedicineFrequency {
-  OD,
-  BD,
-  TDS,
-  QDS
+import android.os.Parcelable
+import androidx.room.TypeConverter
+import com.squareup.moshi.FromJson
+import com.squareup.moshi.ToJson
+import kotlinx.android.parcel.Parcelize
+import org.simple.clinic.util.room.SafeEnumTypeAdapter
+
+sealed class MedicineFrequency : Parcelable {
+
+  @Parcelize
+  object OD : MedicineFrequency() {
+    override fun toString(): String {
+      return "od"
+    }
+  }
+
+  @Parcelize
+  object BD : MedicineFrequency() {
+    override fun toString(): String {
+      return "bd"
+    }
+  }
+
+  @Parcelize
+  object TDS : MedicineFrequency() {
+    override fun toString(): String {
+      return "tds"
+    }
+  }
+
+  @Parcelize
+  object QDS : MedicineFrequency() {
+    override fun toString(): String {
+      return "qds"
+    }
+  }
+
+  @Parcelize
+  data class Unknown(val actualValue: String) : MedicineFrequency()
+
+  object TypeAdapter : SafeEnumTypeAdapter<MedicineFrequency>(
+      knownMappings = mapOf(
+          OD to OD.toString(),
+          BD to BD.toString(),
+          TDS to TDS.toString(),
+          QDS to QDS.toString()
```

Comment:
What I meant was to override `.toString()` to return the same values that they would have as enums ["OD", "BD", "TDS", "QDS"] and leave the rest as is. Sorry if I wasn't clear about this.

Reply:
Ah okay, will change it 👍🏾